plugins/generator-1.18.2/forge-1.18.2/templates/rangeditem/projectile_entity.java.ftl

```diff
@@ -179,6 +183,19 @@ public class ${name}Entity extends AbstractArrow implements ItemSupplier {
 		return entityarrow;
 	}
 
+	<#if data.ammoItem.isEmpty() || data.preserveProjectiles && data.pickupProjectiles>

```

Comment:
You should check for ALL conditions, otherwise projectiles without ammo can be picked up regardless of the pickup setting